tests/InferTypesTest.elm

```diff
@@ -1,106 +1,120 @@
 import AST.Common.Type as Type exposing (Type(..))
 import AST.Typed as Typed
 import Common
-import Common.Types as Types
+import Common.Types as Types exposing (VarName(..))
 import Dict.Any
 import Error exposing (TypeError(..))
 import Expect exposing (Expectation)
+import Fuzz exposing (Fuzzer)
 import Stage.InferTypes
-import Test exposing (Test, describe, test)
-import TestHelpers exposing (located)
+import Test exposing (Test, describe, fuzz, test)
+import TestHelpers exposing (dumpType, located)
 
 
 typeInference : Test
 typeInference =
     let
-        runSection ( description, tests ) =
+        runSection : String -> List ( String, Unwrapped.Expr, Result Error.TypeError String ) -> Test
+        runSection description tests =
             describe description
                 (List.map runTest tests)
 
+        runTest : ( String, Unwrapped.Expr, Result Error.TypeError String ) -> Test
         runTest ( description, input, output ) =
             test description <|
                 \() ->
-                    located input
+                    input
+                        |> Canonical.fromUnwrapped
                         |> Stage.InferTypes.inferExpr
                         |> Result.map Typed.getType
+                        |> Result.map dumpType
                         |> Expect.equal output
     in
     describe "Stage.InferType"
-        (List.map runSection
-            [ ( "list"
-              , [ ( "empty list"
-                  , Canonical.List []
-                  , Ok (List (Var 1))
-                  )
-                , ( "one item"
-                  , Canonical.List
-                        [ located (Canonical.Literal (Literal.Int 1)) ]
-                  , Ok (List Int)
-                  )
-                , ( "more items"
-                  , Canonical.List
-                        [ located (Canonical.Literal (Literal.Int 1))
-                        , located (Canonical.Literal (Literal.Int 2))
-                        , located (Canonical.Literal (Literal.Int 3))
-                        ]
-                  , Ok (List Int)
-                  )
-                , ( "different types"
-                  , Canonical.List
-                        [ located (Canonical.Literal (Literal.Int 1))
-                        , located (Canonical.Literal (Literal.String "2"))
-                        ]
-                  , Err (TypeMismatch Int String)
-                  )
-                , ( "more items with different types"
-                  , Canonical.List
-                        [ located (Canonical.Literal (Literal.Bool True))
-                        , located (Canonical.Literal (Literal.String "two"))
-                        , located (Canonical.Literal (Literal.Int 3))
-                        ]
-                  , Err (TypeMismatch Bool String)
-                  )
-                ]
+        [ runSection "list"
+            [ ( "empty list"
+              , Unwrapped.List []
```

Comment:
`Unwrapped` is a bit non-ideal - "which Unwrapped?"
If no qualifier or `Canonical` is not possible because of collisions, I'd go for `CanonicalU` or something like that.